packages/metro/src/Assets.js

```diff
@@ -177,8 +177,9 @@ async function getAssetData(
   localPath: string,
   assetDataPlugins: $ReadOnlyArray<string>,
   platform: ?string = null,
+  publicPath: ?string = '/assets',
 ): Promise<AssetData> {
-  let assetUrlPath = path.join('/assets', path.dirname(localPath));
+  let assetUrlPath = `${publicPath}/${path.dirname(localPath)}`;
```

Comment:
I think this is causing the end to end snapshot tests to fail

Reply:
Yeah I haven't addressed any tests yet, I need to get around the Server assumptions, but I'm not sure how to handle that.